extension/src/popup/ducks/soroban.ts

```diff
@@ -20,15 +20,18 @@ export const getTokenBalances = createAsyncThunk<
     const tokenIdList = await internalGetTokenIds();
 
     const params = [accountIdentifier(publicKey)];
+    const results = [] as TokenBalances;
 
-    const results = await Promise.all(
-      tokenIdList.map(async (tokenId) => {
-        /*
+    for (let i = 0; i < tokenIdList.length; i += 1) {
+      const tokenId = tokenIdList[i];
+      /*
```

Comment:
I noticed an issue where if you had one token that worked and one token that did not resolve, you weren't getting any balances at all. That's because Promise.all will completely fail if one of the Promises fail. To get around this, I'm simply looping so one token failing does not affect the others